### PR TITLE
[Enhancement](storage) optimize segment compaction log (#14448)

### PR DESCRIPTION
**Pull request by freemandealer**
Signed-off-by: freemandealer <freeman.zhang1992@gmail.com>

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — APPROVED** [review]
LGTM

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -340,16 +338,19 @@ Status BetaRowsetWriter::_do_compact_segments(SegCompactionCandidatesSharedPtr s
...
                    << " _num_segment:" << _num_segment << " _num_segcompacted:" << _num_segcompacted
                    << " list directory:" << fmt::to_string(vlog_buffer);
     }
 
     _segcompacted_point += (end - begin + 1);
 
     uint64_t elapsed = GetCurrentTimeMicros() - begin_time;
-    LOG(INFO) << "BetaRowsetWriter:" << this << " segcompaction completed. elapsed time:" << elapsed
-              << "us. _segcompacted_point update:" << _segcompacted_point;
+    LOG(INFO) << "segcompaction completed. tablet_id:" << _context.tablet_id
```
Comment (luozenglin) [inline be/src/olap/rowset/beta_rowset_writer.cpp:350]: Maybe `fmt::format` is better.

Comment (luozenglin) [reply be/src/olap/rowset/beta_rowset_writer.cpp:350]: Better performance, you can refer to: https://stackoverflow.com/questions/58157033/why-is-fmt-slower-than-stdstringstream

Comment (yiguolei) [reply be/src/olap/rowset/beta_rowset_writer.cpp:350]: I do not think we should use fmt in log... It is very ugly. If it is required, I think it is better to add another logger, like LOG_INFO(fmt, args....);

Comment (freemandealer) [reply be/src/olap/rowset/beta_rowset_writer.cpp:350]: Thanks for explanation. Now I am a little confused at when I should use which ... I will benchmark them in my spare time.

**Pull request merged**